include/protozero/iterators.hpp

```diff
@@ -216,8 +217,73 @@ class const_fixed_iterator {
         return !(*this == rhs);
     }
 
+    const_fixed_iterator& operator--() {
+        m_data -= sizeof(value_type);
+        return *this;
+    }
+
+    const_fixed_iterator operator--(int) {
+        const const_fixed_iterator tmp{*this};
+        --(*this);
+        return tmp;
+    }
+    
+    friend bool operator<(const const_fixed_iterator& lhs, const const_fixed_iterator& rhs) {
```

Comment:
`const_fixed_iterator` is only a single pointer, so this and all other functions like this, should take by value. (It used to have two pointers inside which I changed recently and I forgot to change that for the old functions, too.) These should be `noexcept`, too.